pkg/processor/k8sprocessor/kube/owner_test.go

```diff
@@ -47,6 +48,201 @@ func waitForWatchToBeEstablished(client *fake.Clientset, resource string) <-chan
 	return ch
 }
 
+func Test_OwnerProvider_GetOwners_ReplicaSet(t *testing.T) {
+	c, err := newFakeAPIClientset(k8sconfig.APIConfig{})
+	require.NoError(t, err)
+
+	logger, err := zap.NewDevelopment()
+	require.NoError(t, err)
+
+	op, err := newOwnerProvider(
+		logger,
+		c,
+		labels.Everything(),
+		fields.Everything(),
+		ExtractionRules{
+			Namespace:          true,
+			OwnerLookupEnabled: true,
+			PodUID:             true,
+			PodName:            true,
+			ReplicaSetName:     true,
+			Tags:               NewExtractionFieldTags(),
+		},
+		"kube-system",
+	)
+	require.NoError(t, err)
+
+	client := c.(*fake.Clientset)
+	replicaSetWatchEstablished := waitForWatchToBeEstablished(client, "replicasets")
+
+	op.Start()
+	t.Cleanup(func() {
+		op.Stop()
+	})
+
+	<-replicaSetWatchEstablished
+
+	replicaSet, err := c.AppsV1().ReplicaSets("kube-system").
+		Create(context.Background(),
+			&v1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-rs",
+					Namespace: "kube-system",
+					UID:       "fb9e6935-8936-4959-bd90-4e975a4c2b07",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind: "ReplicaSet",
+				},
+			},
+			metav1.CreateOptions{},
+		)
+	require.NoError(t, err)
+
+	pod := &api_v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-pod",
+			Namespace: "kube-system",
+			UID:       "e98a3d3e-fde9-4b10-8f61-cc37d0357c28",
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Kind: replicaSet.Kind,
+					Name: replicaSet.Name,
+					UID:  replicaSet.UID,
+				},
+			},
+		},
+	}
+
+	_, err = c.CoreV1().Pods("kube-system").
+		Create(context.Background(), pod, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	assert.Eventually(t, func() bool {
+		owners := op.GetOwners(pod)
+		if len(owners) != 1 {
+			t.Logf("owners: %v", owners)
+			return false
+		}
+
+		if uid := owners[0].UID; uid != "fb9e6935-8936-4959-bd90-4e975a4c2b07" {
```

Comment:
Maybe it would be more readable to assign the ReplicaSet to a variable and then use its ID here instead of using magic string?
And similarly in other tests that are added with this PR.

WDYT?

Reply:
OK, added variable names.